calico_containers/pycalico/ipam.py

```diff
@@ -125,8 +136,36 @@ def _get_affine_blocks(self, host, version, pool):
 
         return block_ids
 
-    @handle_errors
-    def _new_affine_block(self, host, version, pool):
+    def _get_pool_affine_blocks(self, pool):
+        """
+        Get the affine blocks and corresponding host for all affine blocks
+        within the specified pool.
+
+        :param pool: Limit blocks to a specific pool,
+        :return: List of tuples (host, cidr)
+        """
+        assert isinstance(pool, IPPool)
+
+        # Construct the path
+        hosts_and_blocks = []
+        try:
+            result = self.etcd_client.read(IPAM_HOSTS_PATH, quorum=True, recursive=True).leaves
+            for child in result:
+                packed = child.key.split("/")
+                if len(packed) == 9:
```

Comment:
I don't really understand what is going on here, but it feels kinda yucky.

What is packed?  Why does it have length 9? What does it mean when it doesn't have length 9?